src/radical/pilot/agent/scheduler/continuous.py

```diff
@@ -267,67 +292,68 @@ def _find_resources(self, node, requested_cores, requested_gpus,
+        alloc_gpus  = 0
 
         if partial:
             # For partial requests the check simplifies: we just check if we
             # have either, some cores *or* gpus *or* local_fs, to serve the
             # request
             if (requested_cores and not free_cores) and \
-                (requested_gpus and not free_gpus) and \
-                (requested_lfs  and not free_lfs):
+               (requested_gpus  and not free_gpus)  and \
+               (requested_lfs   and not free_lfs):
                 return [], [], None
 
             if requested_lfs and \
                 ((requested_cores and not free_cores) and
-                 (requested_gpus and not free_gpus)):
+                 (requested_gpus  and not free_gpus)):
                 return [], [], None
 
         else:
             # For non-partial requests (ie. full requests): its a no-match if
             # either the cpu or gpu request cannot be served.
-            if requested_cores > free_cores or \
-                requested_gpus > free_gpus or \
-                    requested_lfs > free_lfs:
+            if  requested_cores > free_cores or \
+                requested_gpus  > free_gpus  or \
+                requested_lfs   > free_lfs   :
                 return [], [], None
 
         # We can serve the partial or full request - alloc the chunks we need
         # FIXME: chunk gpus, too?
         # We need to land enough procs on a node such that the cores,
         # lfs and gpus requested per proc is available on the same node
 
-        num_procs = list()
+        num_procs = dict()
 
-        if requested_lfs:
-            alloc_lfs = min(requested_lfs, free_lfs)
-            num_procs.append(alloc_lfs / lfs_chunk)
         if requested_cores:
             alloc_cores = min(requested_cores, free_cores)
-            num_procs.append(alloc_cores / core_chunk)
+            num_procs['cores'] = alloc_cores / core_chunk
+
         if requested_gpus:
             alloc_gpus = min(requested_gpus, free_gpus)
-            num_procs.append(alloc_gpus / gpu_chunk)
+            num_procs['gpus'] = alloc_gpus / gpu_chunk
 
-        # Find min number of procs determined across lfs, cores, gpus
-        num_procs = min(num_procs)
-
-        # Find normalized lfs, cores and gpus
         if requested_lfs:
-            alloc_lfs = num_procs * lfs_chunk
-        if requested_cores:
-            alloc_cores = num_procs * core_chunk
-        if requested_gpus:
-            alloc_gpus = num_procs * gpu_chunk
+            alloc_lfs = min(requested_lfs, free_lfs)
+            num_procs['lfs'] = alloc_lfs / lfs_chunk
+
+
+        # Find normalized cores, gpus and lfs
+        if requested_cores: alloc_cores = num_procs['cores'] * core_chunk
+        if requested_gpus : alloc_gpus  = num_procs['gpus']  * gpu_chunk
+        if requested_lfs  : alloc_lfs   = num_procs['lfs']   * lfs_chunk
+        self._log.debug('alc : %s %s %s', alloc_cores, alloc_gpus, alloc_lfs)
```

Comment:
The above was wrong before, for non-uniform processes.  Well, now we agreed that processes *are* uniform, but our CUD does not yet enforce this, so this fix is still needed.

what happened was this:

  - assume we have 2 nodes with 42 cores and  6 gpu each.
  - assume we want to place a unit which asks for 6 cpu process and 12 GPU processes (18 procs in total)
  - assume the nodes have each 9 cores available (18 cores)

We would expect a result like this (a GPU process always also needs a CPU core to run on):

  - node 0:  3 CPU cores + (6 CPU cores + 6 GPUs)
  - node 1:  3 CPU cores + (6 CPU cores + 6 GPUs)

What happened is that the scheduler checked node 0, found space for 3 CPU procs and 6 GPU procs, computed the smalles common denominator (3), and placed 3 CPU procs and 3 (!) GPU procs (1 core + 1 GPU).  3 cores and 3 GPUs were left idle.  The second node was utilized the same way - but the remaining 6 GPU processes could not be placed anymore (out of nodes).

This is fixed now: the least common denominator is not computed anymore, chunking is done per resource type.